### PR TITLE
Updated state management of floating buttons in Interactive book

### LINKED ISSUE #251: Update State Management of Interative Book
**Is your feature request related to a problem? Please describe.**
In the Interactive book, if we scroll, the scroll controller listener functions are rebuilding the whole screen. That is causing lagginess in the app. As an effect, 
- the embedded quiz gets reset every time when we scroll
- some embedded components like Circuit Preview, and Images get reloaded every time the scroll position changes.
- It slows down the app and consumes much higher ram and network bandwidth

**Describe the solution you'd like**
The initial problem that I found is the below code  snippet [of this [file](https://github.com/CircuitVerse/mobile-app/blob/master/lib/ui/views/ib/ib_page_view.dart#L81)]
image
It every time rebuild UI, but actually, a full-screen rebuild is not required, the buttons themselves need to be only rebuilt. with setState, we can not achieve that.
Do, We need to convert the state updates logic from setState to Provider. 

**Are you working on this? (Yes/No)**
Yes

### PR DESCRIPTION
**Pull request by Tanmoy741127**
Fixes #251 

Describe the changes you have made in this PR -
I have updated the state management from tradition setstate to Provider.

This fix solve which things 
- Low memory usage for Interactive book, as the screen will not re-render everytime, just for upate the state of floating buttons
- Circuitverse embeddings loading correctly and no issue while scrolling
- The quiz section will not reset while scroling

Screenshots of the changes (If any) -

A screen recording after updating

https://user-images.githubusercontent.com/57363826/211865395-4056b4af-ee5e-4382-ac2f-8a9031759c5c.mp4

Note: Please check Allow edits from maintainers. if you would like us to assist in the PR.

### CODE REVIEW

**Comment by Tanmoy741127** [comment]
@aman-singh7 @manjotsidhu can you take some time to review this PR.
Thank You

**Comment by Tanmoy741127** [comment]
@aman-singh7 can you check once is the pipeline broken? I can't find any issue with the updated part also it doesn't fail in flutter analyze.

**Comment by aman-singh7** [comment]
> @aman-singh7 can you check once is the pipeline broken? I can't find any issue with the updated part also it doesn't fail in flutter analyze.

Let me see. Are you using the latest flutter version??

File: lib/ui/views/ib/ib_page_view.dart [hunk 1]
```diff
@@ -397,36 +399,45 @@ class _IbPageViewState extends State<IbPageView> {
...
-              disposeOnTap: true,
-              child: const Icon(
-                Icons.arrow_back_rounded,
-                color: IbTheme.primaryColor,
-              ),
-            ),
-          ),
+        ChangeNotifierProvider<IbFloatingButtonState>.value(
+          value: _ibFloatingButtonState,
+          child: Builder(builder: (context) {
```
Comment (aman-singh7) [inline lib/ui/views/ib/ib_page_view.dart:404]: @Tanmoy741127 remove this unnecessary builder widget.

Comment (Tanmoy741127) [reply lib/ui/views/ib/ib_page_view.dart:404]: Okk

**Comment by Tanmoy741127** [comment]
@aman-singh7 can you check it once. I have done the said changes.

**Comment by Tanmoy741127** [comment]
> @aman-singh7 can you check it once. I have done the said changes.

@aman-singh7 if you can get some time, please checkthis PR

**Comment by Tanmoy741127** [comment]
@aman-singh7  @manjotsidhu if you have got some time, please review the PR

**Comment by Tanmoy741127** [comment]
@aman-singh7 @manjotsidhu ig we need to resolve the deprecation warning. else `flutter analyze` is failing

**Comment by manjotsidhu** [comment]
@Tanmoy741127 I have rerun the checks, flutter analyze is still failing

**Comment by Tanmoy741127** [comment]
@manjotsidhu ig I need to merge the code from latest commit

**Comment by manjotsidhu** [comment]
@Tanmoy741127 Yeah, you can rebase your changes against master

**Review by manjotsidhu — APPROVED** [review]

**Pull request merged**